python_modules/dagster/dagster/_core/remote_representation/external_data.py

```diff
@@ -1449,6 +1449,10 @@ def external_asset_nodes_from_defs(
             ):
                 partition_mappings[pk] = partition_mapping
 
+        tags_with_kinds = {
+            **asset_node.tags,
+            **{f"{KIND_PREFIX}{kind}": "" for kind in asset_node.kinds or []},
+        }
```

Comment:
I don't see why this should be necessary. We should be canonicalizing to tags earlier than this

Reply:
Lifted to `AssetSpec` constructor